File: 0526-beautiful-arrangement/0526-beautiful-arrangement.py
class Solution:
    def countArrangement(self, n: int) -> int:
        visited = [False for _ in range(n + 1)]
        count = 0

        def dfs(index):
            nonlocal count
            if index == n + 1:
                count += 1
                return

            for perm in range(1, n + 1):
                if visited[perm] == False and isArrangementBeautiful(perm, index):
                    visited[perm] = True
                    dfs(index + 1)
                    visited[perm] = False

        def isArrangementBeautiful(arrangement, i):
            if arrangement % i == 0 or i % arrangement == 0:
                return True

            return False

        dfs(1)

        return count

File: 0526-beautiful-arrangement/0526-beautiful-arrangement.js
/**
 * @param {number} n
 * @return {number}
 */
var countArrangement = function (n) {
    let count = 0;
    const visited = Array(n + 1).fill(false);

    const dfs = function (index) {
        if (index === n + 1) {
            count++;
            return;
        }

        for (let perm = 1; perm <= n; perm++) {
            if (!visited[perm] && isArrangementBeautiful(perm, index)) {
                visited[perm] = true;
                dfs(index + 1);
                visited[perm] = false;
            }
        }
    }

    dfs(1);

    return count;
};

var isArrangementBeautiful = function (arrangement, i) {
    if (arrangement % i === 0 || i % arrangement === 0) {
        return true;
    }

    return false;
}